### PR TITLE
pythonPackages.watermark: init at 2.2.0

### PR DESCRIPTION
**Pull request by nphilou**
###### Motivation for this change

Useful lib for showing hardware info in notebooks

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/watermark/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+buildPythonPackage rec {
+  pname = "watermark";
+  version = "2.2.0";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "rasbt";
+    repo = pname;
+    rev = "d3553b68dd30ac5b0951a6fae6083236e4c7f3bd";
```
Comment (legendofmiracles) [inline pkgs/development/python-modules/watermark/default.nix:18]: Any reason why you can't use the tag you set in version?

File: pkgs/development/python-modules/watermark/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  ] ++ lib.optionals (pythonOlder "3.8") [ importlib-metadata ];
+
+  checkInputs =  [
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "watermark" ];
+
+  meta = with lib; {
+    description = "IPython extension for printing date and time stamps, version numbers, and hardware information.";
```
Comment (legendofmiracles) [inline pkgs/development/python-modules/watermark/default.nix:33]:
```suggestion
    description = "IPython extension for printing date and time stamps, version numbers, and hardware information";
```

**Comment by legendofmiracles** [comment]
Can you please squash your two commits?

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 144892 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>python38Packages.watermark</li>    <li>python39Packages.watermark</li>  </ul></details>

File: pkgs/development/python-modules/watermark/default.nix (hunk 1 above)
Comment (nphilou) [reply pkgs/development/python-modules/watermark/default.nix:18]: Inattention mostly, I will remember it next time, thank you for pointing it out.

**Comment by jonringer** [comment]
@legendofmiracles sorry, this should have been squashed. I just looked at the diff.

File: pkgs/development/python-modules/watermark/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/watermark/default.nix:18]: Please patch this to make auto updates work.

**Pull request merged**